File: frontend/src/interfaces/Order.ts
import type { User } from "./User";
import type { OrderItem } from "./OrderItem";
import type { Payment } from "./Payment";
import type { OrderPromotion } from "./OrderPromotion";

export interface Order {
  ID: number;
  total_amount: number;
  order_create: string;   // ISO datetime
  order_status: string;   // e.g., "pending" | "paid" | "cancelled"
  user_id: number;
  user?: User;

  // relations
  order_items?: OrderItem[];
  payments?: Payment[];
  order_promotions?: OrderPromotion[];
}

export interface CreateOrderRequest {
  total_amount: number;
  order_status: string;
  user_id: number;
  order_create?: string; // ถ้าไม่ส่ง backend จะใส่ให้
}

export interface UpdateOrderRequest {
  ID: number;
  total_amount?: number;
  order_status?: string;
  order_create?: string;
}

File: backend/entity/order.go
// entity/order.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Order struct {
	gorm.Model
	TotalAmount float64   `json:"total_amount"`
	OrderCreate time.Time `json:"order_create"`
	OrderStatus string    `json:"order_status"`

	UserID uint  `json:"user_id"`
	User   *User `gorm:"foreignKey:UserID" json:"user,omitempty"`

	// relations
	OrderItems      []OrderItem      `gorm:"foreignKey:OrderID" json:"order_items,omitempty"`
	Payments        []Payment        `gorm:"foreignKey:OrderID" json:"payments,omitempty"`
	OrderPromotions []OrderPromotion `gorm:"foreignKey:OrderID" json:"order_promotions,omitempty"`
}
